plugins/BEdita/Core/src/Shell/BeditaShell.php

```diff
@@ -182,19 +235,135 @@ protected function checkDir($dirPath, $label)
+            $this->warn('Unable to connect');
+            $this->err('Error message: ' . $res['error']);
+            if ($this->configModified) {
+                $this->warn('==> Please check your database configuration in config/app.php file');
+                $this->warn("==> See 'DataSources' => 'default' array");
+            }
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Check database connection parameters
+     * On a fresh install placeholders should be found, if placeholders are not found
+     * a manual edit is required
+     *
+     * @return bool True on success, false otherwise
+     */
+    protected function databaseConnection()
+    {
+        $this->configModified = false;
+        $dbParams = Database::basicInfo();
+        $fields = ['host', 'database', 'username', 'password'];
+        foreach ($fields as $name) {
+            $expected = '__BE4_DB_' . strtoupper($name) . '__';
+            if (empty($dbParams[$name]) || $dbParams[$name] !== $expected) {
+                $this->configModified = true;
+            }
+        }
+
+        if ($this->configModified) {
+            $this->warn('Configuration file has been modified!');
+
+            return $this->checkDbConnection();
+        }
+
+        $this->info('A working database connection is needed in order to continue');
+        $this->info('Parameter needed are: host, database, username, password');
+        $res = $this->in('Proceed with setup?', ['y', 'n'], 'n');
+        if ($res != 'y') {
+            $this->info('Database setup stopped');
+
+            return false;
+        }
+
+        $this->dbConnectionUserInput();
+        $dbParams = array_merge($dbParams, $this->userInputData);
+        $dbParams['className'] = 'Cake\Database\Connection';
+        ConnectionManager::config(self::TEMP_SETUP_CFG, $dbParams);
+        ConnectionManager::alias(self::TEMP_SETUP_CFG, 'default');
+
+        return $this->checkDbConnection();
+    }
+
+    /**
+     * Save database connection data in config/app.php file
+     *
+     * @return bool True on success, false otherwise
+     */
+    protected function saveConnectionData()
+    {
+        if (empty($this->configPath)) {
+            $this->configPath = CONFIG . 'app.php';
+        }
+        if (!is_writable($this->configPath)) {
+            $this->warn('Unable to update configuration file');
+            $this->warn('==> Please check write permission on config/app.php file');
+
+            return false;
+        }
+
+        $content = file_get_contents($this->configPath);
+        $fields = ['host', 'database', 'username', 'password'];
+        foreach ($fields as $name) {
+            $placeHolder = '__BE4_DB_' . strtoupper($name) . '__';
+            $content = str_replace($placeHolder, $this->userInputData[$name], $content);
+        }
+
+        // TODO: better php check for $content?
+        $eval = eval(str_replace('<?php', '', $content));
```

Comment:
Maybe we could mitigate the risks doing some validation on data input and using `File\Filesystem\File` operating on files, for example using `http://api.cakephp.org/3.3/class-Cake.Filesystem.File.html#_replaceText.